aggsender/types/certificate_metadata.go

```diff
@@ -35,14 +37,42 @@ type CertificateMetadata struct {
 	CertType uint8 // version >= V2
 }
 
+func (c *CertificateMetadata) String() string {
+	if c == nil {
+		return "CertificateMetadata is nil"
+	}
+	return fmt.Sprintf("Version: %d, FromBlock: %d, Offset: %d, CreatedAt: %d, CertType: %d",
+		c.Version, c.FromBlock, c.Offset, c.CreatedAt, c.CertType)
+}
+
+func (c *CertificateMetadata) BlockRange() (BlockRange, error) {
+	if c == nil {
+		return BlockRangeZero, fmt.Errorf("certificate metadata is nil")
+	}
+	switch c.Version {
+	case CertificateMetadataV0:
+		return BlockRangeZero, fmt.Errorf("cannot get block range from metadata version 0")
+	case CertificateMetadataV1:
+		return NewBlockRange(c.FromBlock, c.FromBlock+uint64(c.Offset)), nil
+	case CertificateMetadataV2:
+		return NewBlockRange(c.FromBlock, c.FromBlock+uint64(c.Offset)), nil
```

Comment:
WDYT, although not insisting to change this, just a suggestion if you find it useful? (maybe slightly less readable, but two lines of code less)
```suggestion
	case CertificateMetadataV1, CertificateMetadataV2:
		return NewBlockRange(c.FromBlock, c.FromBlock+uint64(c.Offset)), nil
```

Reply:
Your suggestion is welcome!,the fewer lines the fewers errors!